BattleTowers/src/main/java/atomicstryker/battletowers/common/AS_WorldGenTower.java

```diff
@@ -381,7 +385,11 @@ public void generate(World world, int ix, int jy, int kz, int towerchoice, boole
                     	}
                     	else
                     	{
-                    		tileentitymobspawner.getSpawnerBaseLogic().setEntityId(new ResourceLocation("battletowers:" + towerChosen.getName() ));
+                    		try{
+                    		Constructor<? extends Event> constructor = (Constructor<? extends Event>) Class.forName("com.EvilNotch.dungeontweeks.main.Events.EventDungeon$Post").getConstructor(TileEntity.class,BlockPos.class,Random.class,ResourceLocation.class,World.class);
+                    		Event event = constructor.newInstance(tileentitymobspawner,tileentitymobspawner.getPos(),world.rand,new ResourceLocation("battletowers:" + towerChosen.getName() ),world);
+                    		MinecraftForge.EVENT_BUS.post(event);
+                    		}catch(Throwable t){t.printStackTrace();}
```

Comment:
Please format these changes to match, there is an eclipse formatter config in my repo

Reply:
Ok working on it will do

Reply:
Ok I couldn't find your formatter but, I did a new commit in the similar format to what the rest of the class looks like hope it's good check it out thanks.